csi/controller.go

```diff
@@ -266,6 +276,17 @@ func (s *OsdCsiServer) CreateVolume(
 		return nil, status.Error(codes.InvalidArgument, e)
 	}
 
+	// Get parent ID from request: snapshot or volume
```

Comment:
A new csi-sanity test checks for this feature. It seems we missed it in the past. This results in a volume snapshot within our driver.

I can change this if needed, but this seemed to make the most sense to me.